VestingSchedule.sol

```diff
@@ -0,0 +1,227 @@
+                bool    revocable,
+                address revoker,
+                address refundDestination,               
+                address vestingContractOwner) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(revoker != address(0), "Revoker is the zero address");
+        require(refundDestination != address(0), "Refund destination is the zero address");
+        // solhint-disable-next-line max-line-length
+        require(vestingCliff <= vestingPeriodSec, "Vesting Cliff is longer than duration");
+        // solhint-disable-next-line max-line-length
+        require(vestingPeriodSec > 0, "Vesting duration is 0");
+        // solhint-disable-next-line max-line-length
+         require(vestingStartTime.add(vestingCliff) > block.timestamp, "Final time is before current time");
+
+        _transferOwnership(vestingContractOwner);
+
+        _beneficiary = beneficiary;
+        _vestingAmount = vestingAmount;
+        _revocable = revocable;
+        _duration = vestingPeriodSec;
+        _cliff = vestingStartTime.add(vestingCliff);
+        _start = vestingStartTime;
+        _refundDestination = refundDestination;
+        _revoker = revoker;
+    }
+
+    /**
+     * @return the beneficiary of the tokens.
+     */
+    function beneficiary() external view returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @return the amount to be vested.
+     */
+    function vestingAmount() external view returns (uint256) {
+        return _vestingAmount;
+    }
+
+    /**
+     * @return the cliff time of the token vesting.
+     */
+    function cliff() external view returns (uint256) {
+        return _cliff;
+    }
+
+    /**
+     * @return the start time of the token vesting.
+     */
+    function start() external view returns (uint256) {
+        return _start;
+    }
+
+    /**
+     * @return the duration of the token vesting.
+     */
+    function duration() external view returns (uint256) {
+        return _duration;
+    }
+
+    /**
+     * @return true if the vesting is revocable.
+     */
+    function revocable() external view returns (bool) {
+        return _revocable;
+    }
+
+    /**
+     * @return the cummulative amount of tokens released.
+     */
+    function released() external view returns (uint256) {
+        return _released;
+    }
+
+    /**
+     * @return true if the vesting is revoked.
+     */
+    function revoked(address token) external view returns (bool) {
+        return _revoked;
+    }
+
+    /**
+     * @return the timestamp of the revocation.
+     */
+    function revokeTime(address token) external view returns (bool) {
+        return _revokeTime;
+    }
+
+    /**
+     * @notice Transfers available released tokens from the vesting back to beneficiary.
+     */
+    function withdraw() external onlyBeneficiary {
```

Comment:
I think we should have a require for `(!_revoked)` too

Reply:
not needed as already discussed as the remaining amount can in fact be withdrawn after the vesting has been revoked